scripts/vim-patch.sh

```diff
@@ -425,8 +425,13 @@ show_vimpatches() {
   get_vim_sources
   printf "\nVim patches missing from Neovim:\n"
 
-  list_missing_vimpatches | while read -r vim_commit; do
-    if [ -n "$(git -C "$VIM_SOURCE_DIR" diff-tree --no-commit-id --name-only  "${vim_commit}" -- runtime)" ]; then
+  local -A runtime_commits
+  for commit in $(git -C "${VIM_SOURCE_DIR}" log --format="%H %D" --decorate=short -- runtime); do
+    runtime_commits[$commit]=1
+  done
+
+  list_missing_vimpatches "$@" | while read -r vim_commit; do
+    if [[ "${runtime_commits[$vim_commit]-}" ]]; then
```

Comment:
let's try to stick with POSIX shell where possible, even if the script is using bashisms the practices tend to bleed out into everything. `if [ -n "$foo" ]` is just as readable.

Reply:
Yeah, while I agree in general, I'd keep it here: as you said it is using an array/map.
And AFAIK "[[" is used / handled by bash directly, while `[` is the actual link to `true` (i.e. external program).

I think for scripts clearly using Bash features, it is OK/good to use `[[` instead of `[`.

Reply:
FWIW, I've tested the `git log --format="%H %D" --decorate=short -- runtime` with Git from Ubuntu Xenial now.